lib/mix/lib/mix/task.ex

```diff
@@ -326,6 +326,17 @@ defmodule Mix.Task do
     end
   end
 
+  @doc false
+  def show_forgotten_apps_warning(apps) do
```

Comment:
I would move this warning to `run_in_apps`. I dropped a comment below. :)

Reply:
This was my first attempt, but I noticed two situations where the warning is displayed multiple times:

* The warning will be displayed for each different task because `run_in_apps` is invoked from `Task.Mix.Do` for each passed task.
* There is a recursive call from the aliases logic (https://github.com/elixir-lang/elixir/pull/11593/files#diff-4b0f94e7f3d2186a8e00b12c6b7c9bda8a7d86e8bb24f8013cf16b405bbdee8aR512), so the warning will be displayed twice in this case, as well.

So, I wanted to make sure the warning happens only once. Note the warning is about the no-exiting apps, so it has to be displayed/controlled at "root level" (in this case, in the context of the initial call from `Task.Mix.Do`).

Does it make sense?

Reply:
Yes, it does. So maybe I would keep it in the `do` task then. I think you can use `Mix.Project.apps_paths` to get the current app names.